components/x-gift-article/src/lib/share-link-actions.js

```diff
@@ -6,29 +6,30 @@ function createMailtoUrl (articleTitle, shareUrl) {
 }
 
 
-function copyToClipboard (copyText) {
-
-	const selected =
-	document.getSelection().rangeCount > 0          // Check if there is any content selected previously
-		? document.getSelection().getRangeAt(0)       // Store selection if found
-		: false;                                      // Mark as false to know no selection existed before
-
-	const el = document.createElement('textarea');  // Create a <textarea> element
-	el.value = copyText;                            // Set its value to the string that you want copied
-	el.setAttribute('readonly', '');                // Make it readonly to be tamper-proof
-	el.style.position = 'absolute';
-	el.style.left = '-9999px';                      // Move outside the screen to make it invisible
-	document.body.appendChild(el);                  // Append the <textarea> element to the HTML document
-
-	el.select();                                    // Select the <textarea> content
-	document.execCommand('copy');                   // Copy
-	document.body.removeChild(el);                  // Remove the <textarea> element
-
-	if (selected) {                                 // If a selection existed before copying
-		document.getSelection().removeAllRanges();    // Unselect everything on the HTML document
-		document.getSelection().addRange(selected);   // Restore the original selection
+function copyToClipboard (targetClass = '.js-gift-article__copy-target') {
+	const inputEl = document.querySelector(targetClass);
```

Comment:
this won't work in the App, where there will usually be three gift article components rendered on the page because of article prerendering. this will select the first `.js-gift-article__copy-target` which will be effectively random. instead, you could pass through `event.target` from the action, and use that to find the surrounding `.js-gift-article__copy-target`? (using `Element.prototype.closest` or similar)

Reply:
@quarterto Thank you for your comment! I always forget about App... 🙈  I will do what you suggested 👍